### PR TITLE
use the branch to tag the docker image and also build team edition

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
update to use the branch name only to make a bit easier to use in mattermod and also make the docker hub a bit cleaner.

also add a stage to build the team edition as well

#### Ticket Link
Part of this Jira: https://mattermost.atlassian.net/browse/MM-18865

### CODE REVIEW

File: build/Jenkinsfile.branch [hunk 1]
```diff
@@ -124,7 +124,7 @@ pipeline {
           withCredentials([usernamePassword(credentialsId: 'matterbuild-docker-hub', usernameVariable: 'DOCKER_USER', passwordVariable: 'DOCKER_PASS')]) {
             sh 'docker login --username ${DOCKER_USER} --password ${DOCKER_PASS}'
             sh """
-            export TAG='${env.BRANCH_NAME}.${env.BUILD_NUMBER}.${SERVER_GIT_COMMIT}'
```
Comment (gabrieljackson) [inline build/Jenkinsfile.branch]: Does this mean we will overwrite previous images from the same branch? I don't think that's an issue, but I wanted to check that it won't affect upgrades or anything.

Comment (cpanato) [reply build/Jenkinsfile.branch]: yes it will override and that is ok

**Pull request merged**